architecture.tex

```diff
@@ -43,16 +46,21 @@ \subsection{Basic Usage}
 
 \subsection{The Core}
 
-The core of a computer algebra system (CAS) refers to the module that is in
-charge of representing symbolic expressions and performing basic manipulations
-with them. In SymPy, every symbolic expression is an instance of properly
-designed Python classes, representing expression trees. The operators are
-represented by the type of an expression and the child nodes are stored in the
-\texttt{args} attribute. A leaf node in the expression tree has an empty
-\texttt{args}.
-The \texttt{args} attribute is provided by the class \texttt{Basic},
-which is a superclass of all SymPy objects and
-provides common methods to all SymPy tree-elements.
+A computer algebra system (CAS) represents mathematical expressions as data
+structures.  For example the mathematical expression $x + y$ is represented as
+a tree with three nodes, $+$, $x$, and $y$, where $x$ and $y$ are ordered
+children of $+$.  As users of the computer algebra system manipulate
+mathematical expressions with traditional mathematical syntax the CAS
+manipulates the underlying data structures.  Automated optimizations and
+computations such as integration, simplification, etc. are all functions that
+consume and produce expression trees.
+
+In SymPy every symbolic expression is an instance of a Python \texttt{Basic}
+class, a superclass of all SymPy types providing common methods to all SymPy
+tree-elements such as traversals, caching, etc..  The children of a node in the
```

Comment:
I didn't see anything at https://www.siam.org/journals/pdf/stylemanual.pdf about this, although the document itself ends several sentences with "etc." with only a single dot. [This](https://english.stackexchange.com/questions/23022/how-to-deal-with-abbreviations-like-etc-at-the-end-of-parentheses-which-are-c) also says to use one dot.